src/openrct2/ride/Vehicle.cpp

```diff
@@ -805,6 +805,21 @@ Vehicle* try_get_vehicle(uint16_t spriteIndex)
     return &sprite->vehicle;
 }
 
+bool rct_sprite::IsVehicle() const
+{
+    return vehicle.sprite_identifier == SPRITE_IDENTIFIER_VEHICLE;
+}
+
+Vehicle* rct_sprite::AsVehicle()
+{
+    Vehicle* result = nullptr;
+    if (IsVehicle())
+    {
+        return reinterpret_cast<Vehicle*>(this);
+    }
+    return result;
```

Comment:
I'd just do a ternary
```suggestion
    return IsVehicle() ? reinterpret_cast<Vehicle*>(this) : nullptr;
```

Reply:
It's going to be removed anyway when i do my is as refactor.